dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -3436,6 +3461,44 @@ void Spdp::SpdpTransport::send_local(const DCPS::MonotonicTimePoint& /*now*/)
   outer_->remove_expired_participants();
 }
 
+void Spdp::SpdpTransport::thread_status_task(const DCPS::MonotonicTimePoint& /*now*/)
```

Comment:
What challenges are there to moving the reporting logic into the Service Participant or at least the DomainParticipantImpl?  That way, it generalizes to all types of discovery.  Alternatively, we could forego the thread status map and have each thing that would be inserting into the map just write its instance when necessary.

I'm not interested is doing this right now but I would like to know if there are certain challenges to doing so.